src/test/java/com/devonfw/sample/archunit/PackageRuleTest.java

```diff
@@ -0,0 +1,76 @@
+package com.devonfw.sample.archunit;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+/**
+ * JUnit test that validates the Packages of this application.
+ */
+
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class PackageRuleTest {
+
+/*Components */
+public static final String COMPONENT_GENERAL = "general";
+
+public static final String COMPONENT_TASK = "task";
+
+/*Layer */
+public static final String LAYER_COMMON = "common";
+
+public static final String LAYER_DATA_ACCESS = "dataaccess";
+
+public static final String LAYER_LOGIC = "logic";
+
+public static final String LAYER_SERVICE = "service";
+
+public static final String LAYER_CLIENT = "client";
+
+public static final String LAYER_BATCH = "batch";
+
+public static final List<String> LAYERS = Arrays.asList(LAYER_CLIENT, LAYER_COMMON, LAYER_DATA_ACCESS,
+LAYER_LOGIC, LAYER_SERVICE, LAYER_BATCH);
+
+/* Pattern */
+private static final String PATTERN_SEGMENT = "[a-zA-Z0-9_]+";
+
+private static final String PATTERN_LAYERS = LAYER_COMMON + "|"
+    + LAYER_DATA_ACCESS + "|" + LAYER_SERVICE  + "|" + LAYER_LOGIC + "|" + LAYER_BATCH + "|"
+    + LAYER_CLIENT;
+
+private static final String COMPONENT_LAYERS = COMPONENT_GENERAL + "|" + COMPONENT_TASK;
+
+public static final String DEFAULT_PATTERN = "(" + PATTERN_SEGMENT + ")\\.(" + COMPONENT_LAYERS + ")\\.(" + PATTERN_LAYERS + ")\\.(" + PATTERN_SEGMENT + ")*";
```

Comment:
The root package typically has multiple segments (e.g. `com.customer.app`).
I would create a separate capturing group for the entire root package as `"(" + PATTERN_SEGMENT + "(\\." + PATTERN_SEGMENT + ")*)".
In case you will actually use capturing groups with according index in the regex code below, a best practice is to add a comment on top of the regex numbering the groups. Example:
https://github.com/devonfw/cobigen/blob/8db8c40c53fe2acedfe15010c69f7fcbee3bce7c/cobigen-plugins/cobigen-templateengines/cobigen-tempeng-agnostic/src/main/java/com/devonfw/cobigen/tempeng/agnostic/AgnosticTemplateEngine.java#L51-L52